docs/agents/agent-development.md

```diff
@@ -237,14 +237,37 @@ Global labels can be specified via the environment variable `ELASTIC_APM_GLOBAL_
 
 ### Transactions
 
+#### Transaction outcome
+
+The optional `outcome` property denotes whether the transaction represents a success or a failure.
+The APM Server converts this to the [`event.outcome`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html) field.
+
+- `"failure"`: Indicates that this transaction describes a failed result.
+- `"success"`: Indicates that this transaction describes a successful result.
+- `"unknown"`:
+  TODO: does unknown make sense in the context of spans and transactions?
+  Would `unknown` apply to an HTTP 202 Accepted? Or a client span that puts a message into a queue?
+  In these cases we only know whether a message could successfully be enqueued but not if it could successfully be processed.
+  Or is this reserved to events that only represent the request part and where the response is a separate event?
+  A transaction contains information about both the request and the response.
+  If `unknown` doesn't apply to transactions it might make more sense to use a boolean as the datatype of this field in the intake API and let APM Server convert to `failure`/`success`. 
+- `null`/`undefined`: There's no information about the outcome of the transaction. \
```

Comment:
I think I would err on the side of leaving this field off. Most spans shouldn't have this defined, in my opinion. I anticipate that this would mostly be present on top-level transactions (which more often have an easy HTTP code to key from).

Reply:
This describes the value for transactions. There's another section for spans. I think that especially for exit spans the outcome is also important to capture.